node: Skip ipcache for remote node IPs if IPsec is enabled

pchaigno opened the pull request:
Before this pull request, if IPsec is enabled, we add all remote node IP addresses to the ipcache of all nodes, regardless of whether `enable-remote-node-identity` is true or false.

This pull request reverts that behavior to only add those IP addresses if remote-node identities, node encryption, or encryption+tunneling are enabled. If encryption+native routing is enabled, we don't need to expose the remote node IP addresses via the ipcache.

christarazi commented on pkg/node/manager/manager.go:
```diff
@@ -318,7 +319,12 @@ func (m *Manager) legacyNodeIpBehavior() bool {
+	// is enabled (to store the SPI for nodes) or if pod encryption is enabled
+	// in tunneling mode (to store the SPI for pod->remote node).
+	return !m.conf.NodeEncryptionEnabled() && !(m.conf.EncryptionEnabled() &&
+		m.conf.TunnelingEnabled()) && !m.conf.RemoteNodeIdentitiesEnabled()
```
Still having a bit of trouble parsing this. Maybe breaking it down into new lines and annotating each line might help? Up to you.

```suggestion
	return !m.conf.NodeEncryptionEnabled() /* ... */
	  && !(m.conf.EncryptionEnabled() && m.conf.TunnelingEnabled()) /* ... */
	  && !m.conf.RemoteNodeIdentitiesEnabled()
```

pchaigno replied:
I split the condition to avoid super long lines. See updated commit.

christarazi reviewed (approved):
LGTM, change makes sense. 

One minor code style nit: now that we've defined `(*DaemonConfig).TunnelingEnabled()`, maybe we could replace all instances of 

```
grep -F ' != option.TunnelDisabled' daemon/
```

with the newly introduced helper, for consistency's sake.

pchaigno committed:
option: Introduce TunnelingEnabled helper

This new method will be used in the following commit to check if
tunneling is enabled from the node package.

Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
node: Skip ipcache for remote node IPs if IPsec is enabled

Before this commit, if IPsec is enabled, we add all remote node IP
addresses to the ipcache of all nodes, regardless of whether
enable-remote-node-identity is true or false.

This commit reverts that behavior to only add those IP addresses if
remote-node identities, node encryption, or encryption+tunneling are
enabled. If encryption+native routing is enabled, we don't need to
expose the remote node IP addresses via the ipcache.

Signed-off-by: Paul Chaignon <paul@cilium.io>
